android/src/main/java/com/tailscale/ipn/ui/viewModel/IpnViewModel.kt

```diff
@@ -124,9 +127,15 @@ open class IpnViewModel : ViewModel() {
   }
 
   fun switchProfile(profile: IpnLocal.LoginProfile, completionHandler: (Result<String>) -> Unit) {
-    Client(viewModelScope).switchProfile(profile) {
-      startVPN()
-      completionHandler(it)
+    val currentUser = loggedInUser.value
+    loggedInUser.set(profile)
```

Comment:
Notifier is really meant to be a channel to get updates from the backend's state. I don't like side-effecting it based on in-progress operations in a view model. Is there a different approach we can take?

Reply:
Here's a clue ...

If you comment out the condition on line 43 of IpnViewModel, then there's no delay in picking up the correct user.

```
viewModelScope.launch {
      Notifier.state.collect {
        // Refresh the user profiles if we're transitioning out of the
        // NeedsLogin state.
        //        if (it == Ipn.State.NeedsLogin) {
        viewModelScope.launch { loadUserProfiles() }
        //        }
      }
    }
 ```

Reply:
Honestly, refreshing the user profile on every state change doesn't seem too bad, so maybe we can just take out the `if (it == Ipn.State.NeedsLogin)` condition?

Reply:
Ack... Done.  That's enough to get it to work.  This should really come in a notifier blob instead of requiring the UI to track it via localAPI calls... but localAPI is cheap enough.